packages/react-components/src/components/dial/dial.tsx

```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+import { DialBase } from './dialBase';
+import { TimeQuery, TimeSeriesData, TimeSeriesDataRequest, StyleSettingsMap, Viewport } from '@iot-app-kit/core';
+import { useTimeSeriesData } from '../../hooks/useTimeSeriesData';
+import { widgetPropertiesFromInputs } from '../../common/widgetPropertiesFromInputs';
+import { Annotations } from '../../common/thresholdTypes';
+import { DialSettings } from './types';
+import { useViewport } from '../../hooks/useViewport';
+
+export const Dial = ({
+  query,
+  viewport: passedInViewport,
+  annotations,
+  styles,
+  settings,
+}: {
+  query: TimeQuery<TimeSeriesData[], TimeSeriesDataRequest>;
+  viewport?: Viewport;
+  annotations?: Annotations;
+  styles?: StyleSettingsMap;
+  settings?: DialSettings;
+}) => {
+  const { dataStreams } = useTimeSeriesData({
+    viewport: passedInViewport,
+    query,
+    settings: { fetchMostRecentBeforeEnd: true },
+    styles,
+  });
+  const { viewport } = useViewport();
+
+  const utilizedViewport = passedInViewport || viewport; // explicitly passed in viewport overrides viewport group
```

Comment:
Should an override be a part of the viewport hook? I imagine every component is going to have this line?

Reply:
I'm not so sure, while your right it'll be a common pattern, it is possibly more intuitive for people to just handle this themselves then have to understand that a viewport passed into useViewport overrides the viewport and makes it so use viewport does nothing